### DISCUSSION TITLE
Camilladsp Has No Config File

### DISCUSSION THREAD

**Discussion by teslasparks301** [rern/rAudio#1703] [General]
I decided to try using camilladsp on my rAudio1 server and it will not start. My error is: 
`CamillaDSP Error: /srv/http/data/camilladsp/configs/camilladsp.yml not valid. `
Upon investigation I found that only the file was missing but so was the path. Am I supposed to create this file? If so, is there a template? Is rAudio supposed to create it?

### _**My particulars:**_
rAudio 20240615
6.6.33-1-rpi aarch64
Raspberry Pi 4B Rev 1.1
BCM2711 • 3.7 GiB
4 x Cortex-A72 @ 1.50 GHz

### **_From Config File:_**
dtoverlay=disable-bt
dtoverlay=disable-wifi
dtoverlay=hifiberry-dacplus
firmware-raspberrypi 20231022-1
linux-firmware 20240610.9c10a208-1
raspberrypi-bootloader 20240612-1

michael

**Comment by rern** [comment]
Please try:
```sh
sed -e '/  Volume:/,/type: Volume/ d
' -e '/- Volume/ d
' /etc/camilladsp/configs/camilladsp.yml > /srv/http/data/camilladsp/configs/camilladsp.yml
```
or the latest image file.

**Reply by teslasparks301** [reply]
Thank you. That did the trick. Now to figure out how to use it. ;)

BTW, the editing did nothing. A diff showed input and output files to be identical.

michael

**Comment by Lightwel** [comment]
Using version 20240710 I have the same problem, but this solution did not work for me.  I am using a 2-channel USB DAC which is recognised by rAudio and works without issue.

System RPi4B 8Gb. 

`[alarm@rAudio ~]$ cat /var/log/camilladsp.log
2024-08-02 16:55:46.217622 INFO  [src/bin.rs:683] CamillaDSP version 2.0.3
2024-08-02 16:55:46.217666 INFO  [src/bin.rs:684] Running on linux, aarch64
2024-08-02 16:55:46.217736 ERROR [src/bin.rs:814] Could not open config file ''. Error: No such file or directory (os error 2)
`
`[alarm@rAudio ~]$ cat /srv/http/data/camilladsp/configs/camilladsp.yml
---
title: ~
description: ~
devices:
  samplerate: 44100
  chunksize: 2048
  queuelimit: 4
  silence_threshold: -60.0
  silence_timeout: 3.0
  capture:
    type: Alsa
    channels: 2
    device: "hw:Loopback,0"
    format: S32LE
  playback:
    type: Alsa
    channels: 2
    device: "hw:0,0"
    format: S16LE
  enable_rate_adjust: false
  target_level: 2048
  adjust_period: 10.0
  resampler: ~
  capture_samplerate: ~
  stop_on_rate_change: false
  rate_measure_interval: 3.0
  volume_ramp_time: ~
mixers: {}
filters:
  Bass:
    type: Biquad
    description: ~
    parameters:
      type: Lowshelf
      freq: 85.0
      q: 0.9
      gain: 0.0
  Treble:
    type: Biquad
    description: ~
    parameters:
      type: Highshelf
      freq: 6500.0
      q: 0.7
      gain: 0.0
processors: ~
pipeline:
  - type: Filter
    channel: 0
    names:
      - Bass
      - Treble
    description: ~
    bypassed: ~
  - type: Filter
    channel: 1
    names:
      - Bass
      - Treble
    description: ~
    bypassed: ~
[alarm@rAudio ~]$ 
`
`[alarm@rAudio ~]$ cat /srv/http/data/mpdconf/conf/camilladsp.conf
audio_output {
        name           "CamillaDSP"
        device         "hw:Loopback,1"
        type           "alsa"
        auto_resample  "no"
        mixer_type     "none"
}
[alarm@rAudio ~]$ 
`

This is my first foray into camilladsp, which I want to use for room correction via REW.

Any help appreciated.

Tks
R

**Comment by Lightwel** [comment]
Sorry about the shouting - must have hit a formatting button without noticing :(
R

**Comment by Lightwel** [comment]
As a double-check I downloaded again the latest release version i20240615 imaged to a new USB and then updated to 20240719.  CamillaDSP seems to be working fine.  Cannot understand the previous problems. 

Best
R
